clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -184,7 +194,7 @@ public abstract T get(long timeout, TimeUnit unit) throws InterruptedException,
      * Returns the result value (or throws any encountered exception) if completed, else returns
      * the given valueIfAbsent.
      */
-    public abstract T getNow(T valueIfAbsent) throws InterruptedException, ExecutionException;
+    public abstract T getNow(T valueIfAbsent) throws ExecutionException;
```

Comment:
Could you please revert this change as you mentioned that it can cause compatible issue.